content/post/practical-introR-TEPHINET.Rmd

```diff
@@ -0,0 +1,682 @@
+
+The R packages are available on the Comprehensive R Archive Network (CRAN) https://cran.r-project.org
+
+Here are the basic commands to use packages:
+
+1. Install a package with the command `install.packages("package-name")`
+2. Use them in R with `library("package-name")`
+
+
+Let's install one of the packages from RECON.
+
+```{r, eval=FALSE}
+install.packages('incidence')
+library(incidence)
+
+```
+
+Library is a directory containing installed packages. We can use `lapply(.libPaths(), dir)` to check which packages are active currently in our R session. 
+
+An important part of a package is the documentation. This is stored in the `vignettes`. To access the basic documentation on a package we can use `browseVignettes("incidence")`
+
+
+
+
+
+# Scoping and Environments
+
+A new environment is created when we create a function. This is important! When we call a function, R first looks for the elements within that function; if the elements do not exist within that function, then R looks for them in the global environment.
+
+- Example of a function in which objects are only available in the global environment
+
+```{r}
+mynewfun <- function() {
+  z = x + y 
+  return(z)
+  
+}
+
+x = 1
+y = 3
+
+mynewfun()
+
+```
+
+
+
+
+- Example of a function in which objects are defined partially in the local environment and partially in the global envoronment
+
+```{r}
+mynewfun <- function(xx) {
+  zz = xx + yy 
+  return(zz)
+  
+}
+
+yy = 4
+mynewfun(xx = 4)
+
+```
+
+This characteristic of R is very important when running any analysis or routine. It is always recommended to NOT use elements within a function that are only available in the global environment.
+
+
+# Create and open data sets
+
+R allows users not only to open but also to create data sets. There are three sources of data sets:
+
+- Data set imported as such (from `.xlsx`, `.csv`, `.stata`, or `.RDS` formats, among many others)
+- Data set that is part of a R package
+- Data set created in our R session
+
+# Tidyverse
+
+In order to better manage data sets, we recommend installing and using the `tidyverse` package, which automatically uploads several packages (dplyr, tidyr, tibble, readr, purr, among others) that are useful for data wrangling.
+
+```{r, echo = TRUE, include=TRUE, eval= TRUE}
+library(tidyverse)
+```
+
+
+
+Let's open and explore a data set imported from Excel
+
+This is the data set from our RECON practical on early outbreak analysis:
+- [PHM-EVD-linelist-2017-10-27.xlsx](../../data/PHM-EVD-linelist-2017-10-27.xlsx):
+
+Let's save this data set in the same directory in which we are currently working.
+
+To import data sets from Excel, we can use the library `readxl`, which is linked to tidyverse. However, we still need to load the `readxl` library, as it is not a core tidyverse package.
+
+```{r, eval=FALSE}
+library(readxl)
+library(here)
+dat <- read_excel(here("static/data/PHM-EVD-linelist-2017-10-27.xlsx"))
```

Comment:
There's no need to have "static" here because the participants won't have the website structure. 

```suggestion
dat <- read_excel(here("data/PHM-EVD-linelist-2017-10-27.xlsx"))
```